src/plugins/WebOfTrust/util/BackgroundJob.java

```diff
@@ -0,0 +1,58 @@
+package plugins.WebOfTrust.util;
+
+/**
+ * A background job that can be triggered to run, for example for performing maintenance jobs.
+ *
+ * Implementations of this interface must ensure that at most one single instance of the background
+ * job is executed at any particular moment, i.e. a single background job must always appear to
+ * be run sequentially. Unless terminated, all triggers must eventually followed by an execution
+ * of the background job.
+ *
+ * @author bertm
+ */
+public interface BackgroundJob {
+    /**
+     * Triggers the execution of the job somewhere in the future. When the background job is
+     * executed is determined by the implementation. Implementations must ensure that this method
+     * is safe to invoke from any thread at any moment.
+     */
+    public void triggerExecution();
+
+    /**
+     * Terminates this background job and prevents future execution of the job. If the job is
+     * currently running, its thread will be {@link Thread#interrupt() interrupted}. Subsequent
+     * invocations of {@link #triggerExecution()} have no effect.
+     * If a background job is already terminated, subsequent invocations of this method have no
+     * effect.
+     * This method does not wait for the job to terminate, it only triggers its termination and
+     * returns immediately. Use this method in conjunction with {@link #waitForTermination(long)}
+     * when blocking behaviour is required.
+     * Implementations must ensure that this method is safe to invoke from any thread at any moment.
+     * @see #isTerminated()
+     * @see #waitForTermination(long)
+     */
+    public void terminate();
+
+    /**
+     * Whether this background job is terminated. If the background job is terminated, invocations
+     * of {@link #triggerExecution()} have no effect. Implementations must ensure that a terminated
+     * background job cannot be restarted, i.e. once this method returns {@code true}, it will
+     * always return {@code true}.
+     * Implementations must ensure that this method is safe to invoke from any thread at any moment.
+     * @return {@code true} if this background job is terminated
+     * @see #terminate()
+     */
+    public boolean isTerminated();
```

Comment:
There are scenarios where it might be useful to check for termination without waiting for it. Unit tests are one example, but there may be other legitimate scenarios in production code. Either way, we provide the developer with an asynchronous action, and there should always be a way to check whether the action has completed already.

I don't see any reason to hide this harmless function, developers who use busy loops should be educated instead of kept from doing so. I will add some Javadoc as you suggested, though phrased a bit less paternalistic.